Exiled.Events/Patches/Events/Scp3114/SlappingPlayer.cs

```diff
@@ -0,0 +1,137 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="SlappingPlayer.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp3114
+{
+    using PlayerRoles.PlayableScps;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using System.Reflection.Emit;
+
+    using Attributes;
+    using Exiled.API.Features.Pools;
+    using Exiled.Events.EventArgs.Scp3114;
+    using Handlers;
+    using HarmonyLib;
+    using PlayerRoles.PlayableScps.Scp3114;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patches <see cref="Scp3114Strangle.ServerProcessCmd" />.
+    ///     Adds the <see cref="Handlers.Scp3114.StranglingFinished" /> and <see cref="Handlers.Scp3114.StranglingFinished" /> event.
+    /// </summary>
+    // ReSharper disable UnusedMember.Local
+    [EventPatch(typeof(Scp3114), nameof(Scp3114.SlappingPlayer))]
+    [HarmonyPatch(typeof(Scp3114Slap), nameof(Scp3114Slap.DamagePlayers))]
+    internal sealed class SlappingPlayer
+    {
+        [HarmonyTranspiler]
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+            Label skipLabel = generator.DefineLabel();
+            var local = generator.DeclareLocal(typeof(SlappingPlayerEventArgs));
+            int index = newInstructions.FindIndex(x => x.Calls(PropertyGetter(typeof(ScpAttackAbilityBase<Scp3114Role>), nameof(ScpAttackAbilityBase<Scp3114Role>.DamageAmount))));
+            int index2 = newInstructions.FindIndex(x => x.opcode == OpCodes.Ldfld && (FieldInfo)x.operand == Field(typeof(Scp3114Slap), nameof(Scp3114Slap._humeShield))) + 2;
+            var injectedInstructions = new CodeInstruction[]
+            {
+                // ev = new SlappingPlayerArgs(this, targetHub, damage amount, hume to regen, true)
+                new(OpCodes.Ldc_R4, 25f),
```

Comment:
Use NW Constant or get it directly

Reply:
The NW constant is only in the code from what I can tell. I can retrieve the instruction at the index if thats what you mean. 

I can find the first index of `ldc.r4` (which defines 25, but may change) ie: `IL_006f: ldc.r4 25`
`
```cs
public override void DamagePlayers()
	{
		Transform playerCameraReference = base.Owner.PlayerCameraReference;
		ReferenceHub primaryTarget = DetectedPlayers.GetPrimaryTarget(playerCameraReference);
		if (!(primaryTarget == null))
		{
			DamagePlayer(primaryTarget, DamageAmount);
			if (HasAttackResultFlag(AttackResult.KilledHuman))
			{
				this.ServerOnKill?.Invoke();
			}
			if (HasAttackResultFlag(AttackResult.AttackedHuman))
			{
				this.ServerOnHit?.Invoke();
			}
			float a = _humeShield.CurValue + 25f; // The constant is not in the class itself, but is embedded in the code.
			_humeShield.CurValue = Mathf.Min(a, _humeShield.MaxValue);
		}
	}
```

Reply:
I fixed it in the way I find it.